engine/src/windowing/window.cpp

```diff
@@ -1,20 +1,21 @@
 // Copyright (c) 2023-present Genesis Engine contributors (see LICENSE.txt)
 
+// internal
 #include "windowing/window.hpp"
-
-#include <format>
 #include "logger/log.hpp"
 
+// external
 #include <GLFW/glfw3.h>
 
+// std
+#include <format>
 #include <cassert>
-#include <utility>
 
 namespace gen
 {
 
-	Window::Window(int width, int height, const char* title)
-		: m_extent{width, height} // NOLINT(cppcoreguidelines-pro-type-member-init)
+	Window::Window(const mim::vec2i& extent, const char* title)
```

Comment:
Nit: `vec2i` is two ints, which will fit in a single 64 bit register anyway, passing by value is fine. Also, in general passing by value is better, even if it takes a bit more memory: compiler and optimizer work much better with function locals than references to external data. Value arguments are also useful with the copy[/move]-and-swap idiom.